062-topic-replication-factor-change.md

```diff
@@ -0,0 +1,254 @@
+A similar error message is reported in Topic Operator logs and `KafkaTopic` resource status:
+
+> Replication factor change not supported, but required for partitions\
+> [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21,\
+> 22, 23, 24]
+
+The only [documented solution](https://strimzi.io/docs/operators/latest/deploying#proc-changing-topic-replicas-str) is to use the `kafka-reassign-partitions.sh` by spinning up a separate interactive pod.
+This is is far from being user friendly, and the user is in charge of doing the calculations to determine the best broker to host a new replica, or which replica needs to be dropped.
+
+When data loss is not a problem, the user can simply delete and recreate the `KafkaTopic` resource with the desired replication factor.
+This is not really an option for most use cases.
+
+## Motivation
+
+The topic replication factor change feature will provide a much better cloud-native experience for Strimzi users.
+They will be able to change the topic replication factor by simply updating the `KafkaTopic` resource, which may also happen through a GitOps pipeline.
+
+This is a non exhaustive list of use cases that will become significantly easier:
+
+- Set a lower replication factor for non-critical topics to reduce the infrastructure costs (compute, storage and network).
+- Set a lower replication factor to deal with resource shortage (trade-off between fault-tolerance and resource utilization).
+- Set a higher replication factor for critical topics to improve durability and fault tolerance.
+
+## Proposal
+
+The replication factor change feature can be implemented by integrating the Topic Operator with Cruise Control.
+The Cluster Operator is already integrated with Cruise Control, so we can reuse some of the existing logic and classes.
+
+At runtime, the Topic Operator watches and periodically reconcile the configuration of all managed and non-paused `KafkaTopic` resources.
+Among the other configurations, the reconciliation logic also search for `.spec.replicas` update by comparing with the actual topic replication factor.
+From the point of view of the Topic Operator, a topic replication factor change can be in the following states:
+
+- **Pending**: `.spec.replicas` is updated, but the replication factor change is not sent to Cruise Control.
+- **Ongoing**: `.spec.replicas` is updated and Cruise Control has accepted the request, but the change is not completed, or there is some error.
+- **Ready**: `.spec.replicas` is updated and Cruise Control execution is completed (no warning in `KafkaTopic` status).
+
+Only one replication factor change can be ongoing at any given time for a specific `KafkaTopic`, additional changes for the same `KafkaTopic` will remain pending, and periodically retried.
+At most `max.active.user.tasks` replication factor changes for different `KafkaTopic` resources can be ongoing at the same time in Cruise Control (5 by default).
```

Comment:
this parameter is not just about replication factor changes (as the sentence above let me think) ... it covers all different types of tasks you can run with Cruise Control (including the most common "rebalancing"). I would make it clearer here.

Reply:
Here I wanted to state what's the maximum RF change that CC can accept by default, but I'll rephrase to avoid any confusion. Thanks.